tapcfg/server.go

```diff
@@ -180,12 +180,12 @@ func genServerConfig(cfg *Config, cfgLogger btclog.Logger,
 		}
 	}
 
-	var hashMailCourier proof.Courier[proof.Recipient]
+	var hashMailCourier proof.Courier
 	if cfg.HashMailCourier != nil &&
-		proofCourierAddr.Scheme == proof.ApertureCourier {
+		proofCourierAddr.Type() == proof.ApertureCourier {
 
 		hashMailBox, err := proof.NewHashMailBox(
-			proofCourierAddr, cfg.HashMailCourier.TlsCertPath,
+			proofCourierAddr.Url(), cfg.HashMailCourier.TlsCertPath,
```

Comment:
It's worth nothing that this'll actually dial and try to establish the TCP connection right here and there: https://github.com/lightninglabs/taproot-assets/blob/ef0f3f1cf9dd7e2dea5131e2b275eef6f1e8bd84/proof/courier.go#L183-L187

I think we instead want to defer this out further, as the hashmail may not even be used at all (in favor of other couriers). Or is this primarily for the itests?

Reply:
I think, as you noted in another comment, this section of code was removed in a subsequent commit.